### DISCUSSION TITLE
Normal AI worker stuck at field ends

### DISCUSSION THREAD

**Discussion by danitbit** [Courseplay/Courseplay_FS22#2332] [General]
When I have the courseplay mod enabled, if I run a normal AI worker on a field and I have that vehicle focused, they will not turn at the end of a field. If I exit the vehicle before they get to the end of the field, the AI turns just fine. If I switch to the vehicle as they are turning, they will not continue to work the field. They get stuck. Is this something that you would like a more detailed bug submission for?

**Comment by Tensuko** [comment]
We don't have much to do with the GIANTS AI Worker.
Why would that be a CP Issue if Giants AI is running?

**Reply by Tensuko** [reply]
CP active in mods, but Giants is driving just fine with me sitting in there.
![fsScreen_2023_02_17_11_46_51](https://user-images.githubusercontent.com/27344744/219624134-274113df-1e00-48f5-8ffd-770c8c95110c.png)
![fsScreen_2023_02_17_11_47_40](https://user-images.githubusercontent.com/27344744/219624433-b5cbba8c-9374-40ff-a505-75a5397b777e.png)

**Reply by danitbit** [reply]
It's a CP problem because it only happens when I'm playing with CP enabled. If I have an AI get stuck on a field, I can go back, disable CP and they'll run just fine. If I go back again and enable CP they get stuck again. Yesterday I did notice that you need multiple tractors, but I will work out a repeatable scenario from a new game later today if you are interested. It happens all of the time to me and it's very frustrating.

**Reply by Tensuko** [reply]
Give me a savegame where I can deproduce this and a logfile.
When I just test it it is all fine and works as expected.

**Reply by danitbit** [reply]
[savegame4.zip](https://github.com/Courseplay/Courseplay_FS22/files/10767938/savegame4.zip)
[log.txt](https://github.com/Courseplay/Courseplay_FS22/files/10767946/log.txt)

The AI seeder will not continue after the end when you're in the tractor.

**Reply by danitbit** [reply]
[savegame5.zip](https://github.com/Courseplay/Courseplay_FS22/files/10767952/savegame5.zip)
[log.txt](https://github.com/Courseplay/Courseplay_FS22/files/10767953/log.txt)

The AI mower will not continue when you are in the tractor

**Reply by danitbit** [reply]
I have used other mods in these saves but they are all disabled except for CP of course. For save 4 the weeder should be bugged out as well.

**Reply by danitbit** [reply]
I have expanded/merged fields in both of those saves. I have not messed with 47 or 46 on Elm Creek though. Not sure if that's related at all.

**Reply by Tensuko** [reply]
Wonder why its working in this direction, even there is something left.
![fsScreen_2023_02_17_18_43_22](https://user-images.githubusercontent.com/27344744/219732572-1bd2c7d7-1eb3-410f-b3b4-d2d21af4258f.png)

**Reply by danitbit** [reply]
I haven't actually tried just going perpendicular when they got stuck. Weird it would work one way but not the other.

**Reply by Tensuko** [reply]
Most strange is, that it only happens when you sit in the vehicle.

**Reply by danitbit** [reply]
Going East at the South end made it get stuck, But East at the North end works. But then it gets stuck again going West.

**Reply by danitbit** [reply]
I'm glad you were able to replicate it through the savegame. Replicating it through experimentation will be a different story, but it happens to all of my saves one way or another.

**Comment by coatsy35** [comment]
I have this also. I thought it was just bad luck with the AI

**Comment by cwattyeso** [comment]
Yeah I thought when Giants released the Platinum Expansion they had messed up or broke something with the AI as that's when I began experiencing the issue. Then I noticed one day after downloading a new release of Courseplay and forgetting to put it into my mod folder, so I launched the game without it installed that the AI worked fine again. I then did some tests with some saves with Courseplay and some without and noticed all the saves with Courseplay installed and active have the issue with AI workers stopping at the end of a row and just sitting there refusing to make their turn and continue if you are riding in the vehicle with them.

**Reply by Tensuko** [reply]
The problem is that we have tried to find the issue and during testing it appeared to be totaly random.
We have no clue why.
Also the Giants debug doesn't give much info expect from turn 0 or 1 and some stuff that on one undestands.
When I find more time I try to get in contact with giants about this problem.
